### PR TITLE
[SPARK-30280][DOC] Update docs for make Hive 2.3 dependency by default

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR update document for make Hive 2.3 dependency by default.

### Why are the changes needed?

The documentation is incorrect.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

File: docs/building-spark.md [hunk 1]
```diff
@@ -83,13 +83,10 @@ Example:
...
 add the `-Phive` and `-Phive-thriftserver` profiles to your existing build options.
-By default, Spark will use Hive 1.2.1 with the `hadoop-2.7` profile, and Hive 2.3.6 with the `hadoop-3.2` profile.
-
-    # With Hive 1.2.1 support
-    ./build/mvn -Pyarn -Phive -Phive-thriftserver -DskipTests clean package
+By default Spark will build with Hive 2.3.6.
 
     # With Hive 2.3.6 support
-    ./build/mvn -Pyarn -Phive -Phive-thriftserver -Phadoop-3.2 -DskipTests clean package
+    ./build/mvn -Pyarn -Phive -Phive-thriftserver -DskipTests clean package
```
Comment (dongjoon-hyun) [inline docs/building-spark.md]: Shall we add the Hive 1.2.1 example back after this for users?
```
# With Hive 1.2.1 support
./build/mvn -Pyarn -Phive -Phive-thriftserver -Phive-1.2 -DskipTests clean package
```

Comment (dongjoon-hyun) [reply docs/building-spark.md]: cc @gatorsmile and @srowen

File: docs/sql-data-sources-hive-tables.md [hunk 2]
```diff
@@ -119,15 +119,15 @@ One of the most important pieces of Spark SQL's Hive support is interaction with
 which enables Spark SQL to access metadata of Hive tables. Starting from Spark 1.4.0, a single binary
 build of Spark SQL can be used to query different versions of Hive metastores, using the configuration described below.
 Note that independent of the version of Hive that is being used to talk to the metastore, internally Spark SQL
-will compile against Hive 1.2.1 and use those classes for internal execution (serdes, UDFs, UDAFs, etc).
+will compile against Hive 2.3.6 and use those classes for internal execution (serdes, UDFs, UDAFs, etc).
```
Comment (srowen) [inline docs/sql-data-sources-hive-tables.md]: How much do we want to commit to a particular version, vs just saying 2.3? I don't feel strongly about it.

File: docs/building-spark.md (hunk 1 above)
Comment (srowen) [reply docs/building-spark.md]: OK either way. I guess we kind of want to discourage Hive 1.x, but doesn't mean it should have a line in the docs still

File: docs/sql-data-sources-hive-tables.md (hunk 2 above)
Comment (wangyum) [reply docs/sql-data-sources-hive-tables.md]: +1 for 2.3.

Comment (dongjoon-hyun) [reply docs/sql-data-sources-hive-tables.md]: This seems to become `built-in Hive` eventually. +1 for that.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you for completing this, @wangyum ! Thank you for your advice, @srowen .

**Pull request closed without merging**